src/software/ai/primitive/direct_wheels_primitive_test.cpp

```diff
@@ -129,10 +129,11 @@ TEST(DirectWheelsPrimTest,
 TEST(DirectWheelsPrimTest,
      test_inequality_operator_with_mismatched_back_right_wheel_power)
 {
+    double mismatched_val = -100.1;
     DirectWheelsPrimitive directwheels_primitive =
         DirectWheelsPrimitive(0, 0, 0, 0, 0, 0.0);
     DirectWheelsPrimitive directwheels_primitive_other =
-        DirectWheelsPrimitive(0, 0, 0, 0, -100.1, 0.0);
+        DirectWheelsPrimitive(0, 0, 0, 0, mismatched_val, 0.0);
```

Comment:
`DirectWheelsPrimitive` takes `int16_t` for the argument for `mismatched_val`, so it's implicitly casting to `int16_t`. Consider fixing this test so we're not passing in a `double` **or** (if you think it's out of scope) at least static_cast it to `int16_t` and file an issue

Reply:
In this case I think declaring a variable for this test is overkill anyways, I prefer the status quo.

Reply:
I think my reasoning was that the test was called `test_inequality_operator_with_mismatched_back_right_wheel_power`. And by mismatched i thought type mismatch. But since its being implicitly casted,  i'll change the parameter to -100.